### PR TITLE
=str failed: FlowSplitWhenSpec

### PR DESCRIPTION
**Pull request by agolubev**
it seems like mapAsync and Sink.ignore works together in a specific way. Adding map broke the test.
So I rewrote it with map and probe to have control over the parent flow.
Let me know if it makes sense.

### CODE REVIEW

**Comment by johanandren** [comment]
Was there a failed issue for this or just a local fail?

**Comment by agolubev** [comment]
sorry, did not put a ref for this: Ref #23199

**Comment by ktoso** [comment]
Thanks, marked as in progress :)

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowSplitWhenSpec.scala [hunk 1]
```diff
@@ -257,14 +259,19 @@ class FlowSplitWhenSpec extends StreamSpec {
...
+      probe.request(1)
+      val future = probe.requestNext()
       an[IllegalStateException] mustBe thrownBy {
-        Await.result(
-          Source.single(1).splitWhen(_ ⇒ true).lift
-            .mapAsync(1) { src ⇒ src.runWith(Sink.ignore).flatMap(_ ⇒ src.runWith(Sink.ignore)) } // Sink.ignore+mapAsync pipes error back
-            .runWith(Sink.ignore),
-          3.seconds)
+        Await.result(future, 3.seconds)
       }
```
Comment (ktoso) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowSplitWhenSpec.scala]: What was the failure we're avoiding here, perhaps it was a real problem interaction?
(not sure if test was wrong or impl is wrong)

Comment (agolubev) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowSplitWhenSpec.scala]: I could not reproduce it at the first place. However, I changed mapAsync to map and test failed. The reason was that stream is finished at a time when the second materialization failed substream so no exception was propagated. My best guess was that there is no issue with code (until you think differently) but with test itself - so I rewrote it to have the stream wait.

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by agolubev** [comment]
looking for another LGTM

**Review by johanandren — APPROVED** [review]
LGTM, thanks

**Comment by agolubev** [comment]
Thanks!

**Pull request merged**